### ISSUE TITLE
Automatically-detected indirect jump table has underestimated length

### ISSUE THREAD

**Issue by ExpHP** [Vector35/binaryninja-api#1547]
Version 1.3.2015 Personal (Build ID 90ff0608)

Seen in many places in the binary I am looking at.  A function will have a pair of a jumptable and index table, where the index table is longer than the jumptable; but binja will assign them both the same length (and usually detects a function where it believes the index table ends).

Contact me on the Slack (ExpHP) for the database file.

![image](https://user-images.githubusercontent.com/1411280/76248006-db85c100-6216-11ea-8286-1c5739b84f38.png)

**Comment by ExpHP** [comment]
I went through my binary and manually fixed all of the lengths on my own end, recording the results as I went along.

In each case, I double checked the apparent length (based on the values of the bytes) against the code before the jump to see if there was an obvious bound check (one was found in all cases for this binary).  Here's what I believe to be a complete list of indirect jumptable addresses and correct lengths:

```
0x414cd0 : 128
0x422d70 : 702
0x4240c8 : 63
0x424de4 : 81
0x432604 : 16
0x4392ac : 16
0x43d130 : 15  (also thanks to this bug, a function start is missed at 43d140)
0x45a6c8 : 24
0x45f758 : 36
0x45f870 : 32
0x45f91c : 30
0x462d8c : 215
0x462fa8 : 313
0x463b14 : 17
0x468bac : 28
0x46c890 : 30
0x473b58 : 94
```

**Comment by bpotchik** [comment]
Fixed in `2.2.2629-dev`.
